### DISCUSSION TITLE
$sql->update() in before/afterUpdate

### DISCUSSION THREAD

**Discussion by LaocheXe** [e107inc/e107#5057] [Q&A Plugins] [answered]
So my plugin allows the admin to add a vehicle to the database, when doing so they are also creating a new media category (core_media_cat) by using $sql->insert(); in afterCreate function in the plugin admin_config.php... That works. What I can't get to work is the $sql->update(); when I'm editing/updating the vehicle.

The code I have is:

$sql = e107::getDb();

$type = 'image_';
$oldFiveM = $old_data['v_fmname'];
$old_catGor = 'dealership_'.$type.$oldFiveM;

$fivem = $new_data['v_fmname'];
$catGor = 'dealership_'.$type.$fivem;

$updateSQL = array(
	'media_cat_owner' => 'dealership',
	'media_cat_category' => $catGor,
	'media_cat_title' => $new_data['v_name'],
	'media_cat_sef' => $new_data['v_sef'],
	'media_cat_class' => 0,
	'media_cat_image' => $new_data['v_thumb'],
	'WHERE' => 'media_cat_category = '.$old_catGor.''
);

$sql->update('core_media_cat', $updateSQL);
return true;
--------------------------

No matter how many times I go into the admin area to change the name/title or anything else - nothing updates. I am confused, and following the Developer documentation, I can't figure out why it's not working. I've even went, and looked at the new dev doc which added 'data' => array() to my code, and still no update.

**Comment by tgtje** [comment]
@LaocheXe  again outside 'look', did you check if log files were created and written into the database? just an idea... maybe something is missing and needs to be added (unknown terrain) do read admin/update routine file..iff it has pointers?? success

**Reply by LaocheXe** [reply]
Thanks @tgtje looks like update can't be used like insert - had to change my method. removed the array, and just did it simple:

$updateSQL = "media_cat_category = '".$catGor."', media_cat_title = '".$new_data['v_name']."', media_cat_sef = '".$new_data['v_sef']."', media_cat_image = '".$new_data['v_thumb']."' WHERE media_cat_category = '".$old_catGor."'";

Notice this when looking through e107 files for $sql->update. I believe the method shown on e107 dev documents might be wrong.

**Reply by tgtje** [reply]
Could be, but i know there was a start to rewrite things, damn, lost the link. Moc was mostly busy with it.

**Comment by Jimmi08** [comment, marked as answer]
Just to have it in one place: 
https://github.com/e107inc/e107/issues/4138
https://github.com/e107inc/e107/issues/3740
https://github.com/e107inc/e107/issues/3763

documentation for update() is wrong,  
 
This was correct answer:
https://github.com/e107inc/e107/issues/3740#issuecomment-476824605

Maybe in new Moc documentation it is fixed.  

Sorry for short report, limited time for answer.

**Reply by Moc** [reply]
Jimmi is right. This should work. See https://github.com/e107inc/e107/issues/3740#issuecomment-476824605

https://devguide.e107.org/classes-and-methods/database#update

This documentation is not completed yet,  but does contain the latest information.
